File: silver/25496/python/main.py

if (__name__ == "__main__"):
    p, n = map(int, input().split())
    a = list(map(int, input().split()))[:n]
    a.sort()

    index = 0
    while (p < 200):
        if (index == n): break
        p += a[index]
        index += 1
    print(index)

File: silver/25496/c++/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main()
{
    int n, p;
    std::cin >> p >> n;
    std::vector<int> arr(n);
    
    for (int i = 0; i < n; i++) {
        std::cin >> arr[i];
    }

    std::sort(arr.begin(), arr.end());

    int index = 0;
    while (p < 200) {
        if (index == n) {
            break;
        }
        p += arr[index++];
    }

    std::cout << index;

    return 0;
}